src/scheduler/workers/processListsBeforeAndDuringStart/main.ts

```diff
@@ -1,55 +1,54 @@
 import { endOfDay, isSameDay, isWithinInterval, startOfDay, subDays } from "date-fns";
 import { lowerCase, startCase } from "lodash";
 import { AuditEvent, ListItemEvent } from "@prisma/client";
+import type { Event, Audit } from "@prisma/client";
+import { prisma } from "scheduler/prismaClient";
+import type { SendEmailResponse } from "notifications-node-client";
 
 import { findListItems, findListsWithCurrentAnnualReview, updateIsAnnualReview } from "scheduler/dbHelpers";
 import { logger } from "scheduler/logger";
 import type {
-  Audit,
   List,
   ListAnnualReviewPostReminderType,
   ListItemAnnualReviewProviderReminderType,
   ListItemWithHistory,
 } from "shared/types";
 import type { MilestoneTillAnnualReview } from "../../batch/helpers";
-import { formatDate, isEmailSentBefore } from "./helpers";
+import { formatDate, hasDataContainingReference } from "./helpers";
 import { createAnnualReviewProviderUrl } from "scheduler/workers/createAnnualReviewProviderUrl";
 import {
   sendAnnualReviewPostEmail,
   sendAnnualReviewProviderEmail,
 } from "scheduler/workers/processListsBeforeAndDuringStart/govukNotify";
-import { findAuditEvents } from "./audit";
-import { recordListItemEvent } from "shared/audit";
+import { findAllReminderEvents, findAllReminderAudits } from "./audit";
 import type { BaseDeserialisedWebhookData } from "shared/deserialiserTypes";
+import { addReminderEvent } from "../helpers/addReminderEvent";
 
 async function processPostEmailsForList(
   list: List,
   milestoneTillAnnualReview: MilestoneTillAnnualReview,
   reminderType: ListAnnualReviewPostReminderType | ListItemAnnualReviewProviderReminderType
 ) {
   // Check if sent before
-  let emailSent = false;
   if (!list.jsonData.users) {
     logger.info(
       `Unable to send email to post for ${milestoneTillAnnualReview}. No users identified for List ${list.id}.`
     );
     return;
   }
-  for (const publisherEmail of list.jsonData.users) {
-    const { result } = await sendAnnualReviewPostEmail(
+
+  const postEmailPromises = list.jsonData.users.map(async (publisherEmail) => {
+    return await sendAnnualReviewPostEmail(
       milestoneTillAnnualReview,
       publisherEmail,
       lowerCase(startCase(list.type)),
       list?.country?.name ?? "",
       formatDate(list.nextAnnualReviewStartDate)
     );
-    if (!emailSent && result) {
-      emailSent = result;
-    }
-  }
-  // @todo the following code would be used if using Promise.allSettled
-  // const sendResult = await Promise.allSettled(postEmailPromises);
-  // const emailSent = sendResult.find((result) => result.status === "fulfilled" && result.value);
+  });
+
+  const sendResult = await Promise.allSettled(postEmailPromises);
+  const emailSent = sendResult.find((result) => result.status === "fulfilled" && result.value);
```

Comment:
Maybe `Promise.any` would be a better fit here actually. `allSettled` would be better if we need to know each of the responses, but it looks like this code only cares about the first success it finds. 

```suggestion
 try { 
  const emailSent = await Promise.any(postEmailPromises);
  // create the audit - emailSent will the the first request that resolved.
 } catch (e) {
  // if none of the emails sent
 }
```